SystemServices/SystemServices.cpp

```diff
@@ -3761,55 +3759,192 @@ namespace WPEFramework {
                 JsonObject& response)
         {
             bool status = false;
-            string src, value;
-            WakeupSrcType_t srcType;
-            bool config;
-            int paramErr = 0;
-            IARM_Bus_PWRMgr_SetWakeupSrcConfig_Param_t param;
-            if (parameters.HasLabel("wakeupSrc") && parameters.HasLabel("config")) {
-                src = parameters["wakeupSrc"].String();
-                srcType = (WakeupSrcType_t)atoi(src.c_str());
-                value = parameters["config"].String();
-                config = (bool)atoi(value.c_str());
-
-                switch(srcType){
-                    case WAKEUPSRC_VOICE:
-                    case WAKEUPSRC_PRESENCE_DETECTION:
-                    case WAKEUPSRC_BLUETOOTH:
-                    case WAKEUPSRC_WIFI:
-                    case WAKEUPSRC_IR:
-                    case WAKEUPSRC_POWER_KEY:
-                    case WAKEUPSRC_TIMER:
-                    case WAKEUPSRC_CEC:
-                    case WAKEUPSRC_LAN:
-                        param.srcType = srcType;
-                        param.config = config;
-                        break;
-                    default:
-                        LOGERR("setWakeupSrcConfiguration invalid parameter\n");
-                        status = false;
-                        paramErr = 1;
+            unsigned int srcType = 0x0;
+            unsigned int config = 0x0;
+            unsigned int powerState = 0x0;
+            LOGWARN(" %s: %d Entry srcType:%x  config :%x \n",__FUNCTION__,__LINE__,srcType ,config);
+            if (parameters.HasLabel("powerState")) 
+            {
+                string state = parameters["powerState"].String();
+                if(!state.compare("LIGHT_SLEEP"))
+                    powerState = 1 << IARM_BUS_PWRMGR_POWERSTATE_STANDBY_LIGHT_SLEEP;
+                if(!state.compare("DEEP_SLEEP"))
+                    powerState =  1 << IARM_BUS_PWRMGR_POWERSTATE_STANDBY_DEEP_SLEEP;
+                if(!state.compare("DEFAULT"))
+                    powerState = (1<<IARM_BUS_PWRMGR_POWERSTATE_STANDBY_LIGHT_SLEEP) | \
+                                             (1 << IARM_BUS_PWRMGR_POWERSTATE_STANDBY_DEEP_SLEEP);
+            }
+            else{
+                    powerState = (1<<IARM_BUS_PWRMGR_POWERSTATE_STANDBY_LIGHT_SLEEP) | \
+                                             (1 << IARM_BUS_PWRMGR_POWERSTATE_STANDBY_DEEP_SLEEP);
+            }
+            LOGWARN("%s: %d Power State stored:%x \r\n",__FUNCTION__,__LINE__,powerState);
+            if (parameters.HasLabel("WAKEUPSRC_VOICE")) 
+            {
+                srcType |= 1<<WAKEUPSRC_VOICE;
+                if(parameters["WAKEUPSRC_VOICE"].Boolean())
+                    config |= 1<<WAKEUPSRC_VOICE;
+                LOGWARN("%s: %d WAKEUPSRC_VOICE srcType:%x  config :%x \n",__FUNCTION__,__LINE__,srcType ,config);
```

Comment:
could you move this at end as it is repeated in every if else

Reply:
removed log lines to common place